# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/webauthn/WebAuthnManager.h
# Commit: d28545793e92
# Full Hash: d28545793e92c94cf8355690574ea578e3e8b1c9
# Author: J.C. Jones <jjones@mozilla.com>
# Date: 2019-04-02 08:35:12
# Regressor Bug: 1448408
# File Overlap Count: 2
# Description:
#   Bug 1540378 - Web Authentication: Fix teardown during cycle collection r=keeler,mccr8
#   
#   In Bug 1448408 ("Don't listen to visibility events"), it became possible to
#   close a tab without a visibility event to cause transactions to cancel. This
#   is a longstanding bug that was covered up by the visibility events. This patch
# ==============================================================================

diff -r dccd58970fb0 -r d28545793e92 dom/webauthn/WebAuthnManager.h
--- a/dom/webauthn/WebAuthnManager.h	Tue Apr 02 00:30:57 2019 +0000
+++ b/dom/webauthn/WebAuthnManager.h	Mon Apr 01 23:13:26 2019 +0000
@@ -116,10 +116,11 @@
  private:
   virtual ~WebAuthnManager();
 
+  // Rejects the current transaction and calls ClearTransaction().
+  void RejectTransaction(const nsresult& aError);
+
   // Clears all information we have about the current transaction.
   void ClearTransaction();
-  // Rejects the current transaction and calls ClearTransaction().
-  void RejectTransaction(const nsresult& aError);
 
   // The current transaction, if any.
   Maybe<WebAuthnTransaction> mTransaction;
